xds/internal/balancer/cdsbalancer/cluster_handler.go

```diff
@@ -176,62 +162,74 @@ func (c *clusterNode) handleResp(clusterUpdate xdsclient.ClusterUpdate, err erro
 		return
 	}
 
-	// deletedChild helps determine whether this callback will ping the overall clusterHandler to try and construct an
+	// deltaInClusterUpdateFields determines whether there was a delta in the clusterUpdate fields (forgetting the children).
+	// This will be used to help determine whether to pingClusterHandler at the end of this callback or not.
+	deltaInClusterUpdateFields := clusterUpdate.ClusterName != c.clusterUpdate.ClusterName || clusterUpdate.ClusterType != c.clusterUpdate.ClusterType
+	c.receivedUpdate = true
+	c.clusterUpdate = clusterUpdate
+
+	// If the cluster was a leaf node, if the cluster update received had change in the cluster update then the overall
+	// cluster update would change and there is a possibility for the overall update to build so ping cluster handler to
+	// return. Also, if there was any children from previously, delete the children, as the cluster type is no longer an
+	// aggregate cluster.
+	if clusterUpdate.ClusterType != xdsclient.ClusterTypeAggregate {
+		for _, child := range c.children {
```

Comment:
You also need to clear `c.children = nil`.

Reply:
Added after for loop.